pkg/skaffold/test/test.go

```diff
@@ -113,19 +113,28 @@ func (t FullTester) Test(ctx context.Context, out io.Writer, bRes []build.Artifa
 
 func (t FullTester) runTests(ctx context.Context, out io.Writer, bRes []build.Artifact) error {
 	for _, test := range t.testCases {
-		if err := t.runStructureTests(ctx, out, bRes, test); err != nil {
-			return fmt.Errorf("running structure tests: %w", err)
+		if len(test.CustomTests) != 0 {
+			if err := t.runCustomTests(ctx, out, bRes, test); err != nil {
+				return fmt.Errorf("running custom tests: %w", err)
+			}
+		}
+		if len(test.StructureTests) != 0 {
+			if err := t.runStructureTests(ctx, out, bRes, test); err != nil {
+				return fmt.Errorf("running structure tests: %w", err)
+			}
 		}
 	}
 
 	return nil
 }
 
-func (t FullTester) runStructureTests(ctx context.Context, out io.Writer, bRes []build.Artifact, tc *latest.TestCase) error {
-	if len(tc.StructureTests) == 0 {
-		return nil
-	}
+func (t FullTester) runCustomTests(ctx context.Context, out io.Writer, bRes []build.Artifact, tc *latest.TestCase) error {
+	// Not implemented yet
+	color.Red.Fprintln(out, "Custom tests not implemented yet...")
+	return fmt.Errorf("custom tests not implemented yet")
```

Comment:
since this PR introduces the config changes that's user-facing, it should ideally be merged *after* this function is implemented so as not to expose incomplete features to the user. Or else be implemented in the same PR.

Reply:
Implemented the per feedback for @tejal29 to finalize the config changes & unblock on implementation.

Reply:
um. There is some confusion.
My feedback was to implement the classes first with tests and then add the config change. 
In this case, 
add the implementation for `runCustomTests` with the above signature. 
Add tests for `runCustomTests` and test it out.

Reply:
@nkubala Please provide your input.

Reply:
as per our discussion out of band, let's merge this in tandem with the implementation so we're not exposing anything incomplete.